osu.Game/Graphics/UserInterface/OsuTextBox.cs

```diff
@@ -59,5 +58,21 @@ protected override void OnFocusLost(InputState state)
         }
 
         protected override Drawable GetDrawableCharacter(char c) => new OsuSpriteText { Text = c.ToString(), TextSize = CalculatedTextSize };
+
+        public bool OnPressed(GlobalAction action)
+        {
+            if (action == GlobalAction.Back)
```

Comment:
I don't think this is working how you want this to work. Pressing esc in song select should delete the search text if there is any, but currently it's booting the user straight to menu.

Also, why is this in `OsuTextBox`? Surely we don't desire this functionality in every single textbox game wide? This means that pressing escape in the chat overlay would delete text? I can't actually find a case where this is being triggered correctly right now.

Reply:
Yeah, I intentionally moved this to standardise behaviour for the time being. If you disagree I guess we can move it back to FocusedTextBox